runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -4754,6 +4927,28 @@ private String16FW offsetMetadataToString(
             offsetMetadata.offset(), offsetMetadata.limit()));
     }
 
+    public static final class PublishClientMetadata
```

Comment:
If the state is being shared across different two stream factories, then the class representing the state should be defined outside of both stream factories to make the non-encapsulation clear.